protocols/upnp/src/provider.rs

```diff
@@ -18,15 +18,74 @@
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 // DEALINGS IN THE SOFTWARE.
 
-use std::{error::Error, net::IpAddr};
+use std::{error::Error, fmt, net};
 
-use crate::{
-    behaviour::{GatewayEvent, GatewayRequest},
-    Config,
-};
+use crate::behaviour::{GatewayEvent, GatewayRequest};
 use async_trait::async_trait;
 use futures::channel::mpsc::{Receiver, Sender};
 
+#[derive(Clone, Copy, Debug)]
+pub(crate) struct IpAddr(pub(crate) net::IpAddr);
```

Comment:
I think we could avoid this by just having `is_global` as a free function instead of taking `self`. Saves a bit of code and wrapping.

Reply:
yeah, makes sense. Updated, thanks!